[4.0] Active menu does not exist, fatal error

sakiss opened the pull request:
Pull Request for Issue # .
https://github.com/joomla/joomla-cms/issues/30812

### Summary of Changes
`$app->getMenu()->getActive();`
can return a _MenuItem_ or null

by calling a function when null is returned, it generates a fatal error.


### Testing Instructions
Add a dummy/non existent ItemId in your non sef urls.
E.g. _/index.php?option=com_content&view=article&id=2:another-article&catid=9&itemId=5000_
In that case there is no menu item with id 5000 

### Actual result BEFORE applying this Pull Request
![Screenshot_joomla_missing_menu_ietm](https://user-images.githubusercontent.com/1838041/96142871-3a2cbf00-0f0b-11eb-92bc-5507135f67a7.png)


### Expected result AFTER applying this Pull Request
No fatal error


### Documentation Changes Required
No

infograf768 committed:
seperating multiple messages with border top

laoneo committed:
[4.0] Ignore null results in events (#30408)

richard67 committed:
[4.0] Move admin module "mod_version" to position "status" on update from 3.10 (#30383)

* Move mod_version to position "status" on update from 3.10

* Revert "Move mod_version to position "status" on update from 3.10"

This reverts commit 71719e13964696f4c2565fdfb8d8f237724c744f.

* Move mod_version to the right place when updating from 3.10

* SQL CS even if I don't like this but it seems to be commonly used

infograf768 committed:
[4.0] Fix debuguser/debuggroup icon color (#30414)

Commit:
[4.0] Hide language column if not multilanguage in modal (#30344)

* Check multilanguage setting to display language column

* Optimize code

* Apply check to Contacts and Newsfeeds

* Add multilanguage check

* Check frontend too

* Revert

infograf768 committed:
[4.0] Installation: Adding a hint on top of Password field (#30402)

* [4.0] Installation: Adding a hint in Password field

* fix accessibility

* Added small

Commit:
Remove coloring from icons (#30415)

infograf768 committed:
[4.0] Fix multilingual status icons colors (#30411)

Commit:
Increase minimum length

Commit:
Fix meter displaying complete message prematurely

Commit:
Add strenthmeter attribute to installation

infograf768 committed:
[4.0] Fix Safari issue with overflow hidden (#30420)

* [4.0] Fix Safari issue with overflow hidden

* correcting scss format

* Another solution adapted to various versions of Safari

* scss cs

* scss cs

* last scss cs...

Commit:
Change meter values to display green when password requirements are met

infograf768 committed:
[4.0] {Cassiopea] Implementing Password length/meter

infograf768 committed:
[4.0] Fix Fieldset description display (#30435)

infograf768 committed:
[4.0] Url Language Code is not a prefix (#30440)

* [4.0] URL LANGUAGE CODE is not a prefix

* modified desc as suggested

infograf768 committed:
[4.0] Implementing display of fieldset descriptions for fieldset
children

infograf768 committed:
Taking off description

infograf768 committed:
scss cs

infograf768 committed:
use @Quy suggestion

infograf768 committed:
file cs

dgrammatiko committed:
[4.0] Media field fix (#30455)

* fix

* typo, (well nothing new)

* Allow media fields in subforms

- Cleanup Media field
- Expose initilazation method of Bootstrpa Modlas to a Joomla.Bootstrap object
- Initilise Modlas on the lifecycle of the Custom Element

* Paleolithic js mode only in the legacy…

* CS

* Proper check

* Pffff

* Update joomla-field-media.w-c.es6.js

N6REJ committed:
[4.0] Convert icon used for jgrid defaults to fas fa- (#30464)

* make icon-circle use fa-circle

* allows for including icon- in call

N6REJ committed:
move fa-fw to last class to fit convention (#30469)

brianteeman committed:
[4.0] Long labels wrapping (#30474)

When we have fields displayed in columns with the label displayed above the input there is a width limit of 240px on the label.

This results in long labels wrapping when it is not needed as the column is wider than 240px.

In english we dont see this is many places but the easiest is in the __configure edit screen_  fields

This simple PR changes the label width to 100% when the fields are displayed in columns like this.

As this is an scss change you will need to `node build.js --compile-css` in order to test

### Before

### After

chmst committed:
Fix mod_articles_latest (#30459)

drmenzelit committed:
[4.0] Add a parameter for "back-to-top" button in Cassiopeia (#30441)

* [4.0] Add a parameter for back-to-top button in Cassiopeia

Added a parameter to show / hide the back-to-top button in Cassiopeia.
Currently the footer appears when a module has been published on position "footer". And the back-to-top only appears if the footer is visible. Since the footer is part of the grid, it is not easy to move the back-to-top outside the grid.

Now it is possible to decide to have a back-to-top or not. If yes, the footer will be rendered showing the button, independently if there is a module in the position footer.
It is also possible to show only modules (switch back-to-top off) or modules and back-to-top (switch on).

* Added smooth scrolling

Add smooth scrolling taking in account the preferences of the user (reduced motion).
Improve  language file.

* Correct height and position of back-to-top button

* Correct code style

* Correct code style 2

* Change smooth scrolling behavior to take in account reduced motion preferences

* Language file corrected, language string changed to match others

* Update language/en-GB/tpl_cassiopeia.ini

Link instead button

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update templates/cassiopeia/scss/blocks/_global.scss

Comment for smooth scroll

Co-authored-by: Brian Teeman <brian@teeman.net>

Co-authored-by: Brian Teeman <brian@teeman.net>

N6REJ committed:
fix improper family name. (#30481)

brianteeman committed:
[4.0] update helptoc (#30490)

* [4.0] update helptoc

Removes the string for the expired cache page that no longer exists and removes it from the index displayed on the left hand side in the help page

* api

zero-24 committed:
[4.0] Add new permissions-policy to the HTTPHeaders Plugin (#30491)

* add permissions-policy

* add Permissions-Policy to the dropdown

* alpha order

Commit:
[4.0] Error when changing status of tagged content items (#30466)

* Correct column name

* Fix changing UCM state

brianteeman committed:
[4.0] br tag (#30503)

* [4.0] br tag

In Joomla 4 we use `<br>` not `<br />`

* bad grep

brianteeman committed:
[4.0] Remove old search component (#30506)

* [4.0] Remove old search component

The regular search component is not in J4

This pr removes reference to it in the help system
`administrator/index.php?option=com_admin&view=help`

* revert

brianteeman committed:
[4.0] Help Dashboard (#30508)

* [4.0] Help Dashboard

This PR makes multiple changes to the help dashboard. In order of importance they are.

- Created a new section "Start Here"
- Moved "Joomla Help" to the "Start Here" section to give it greater priority as it really is the first place you should look for help
- Removed target=_blank from "Joomla Help"
- Moved "Documentation Wiki" from "Resources" to the top of "Additional Help"

* deduplicate

* compass icon

Hackwar committed:
Removing patch installation (already in the docker image)
Removing cache steps for a composer cache

Hackwar committed:
Removing outdated dependency

laoneo committed:
[4.0] Check if orderedby element is available in searchtools (#30428)

* Check if orderedby element is available in searchtools

* Do not parse the dom twice

laoneo committed:
When there is no filter form the admin list view should not crash (#30307)

hans2103 committed:
Update _icomoon.scss (#30436)

maliouris committed:
Clear checkboxes on back button (#30498)

Co-authored-by: Ioannis Maliouris <giannism@e-avenue.eu>

Commit:
Sort options (#30531)

Fedik committed:
[4.0] FieldsHelper: Choose a first available category  correctly (#30268)

infograf768 committed:
[4.0] Implementing display of password requirements for frontend (#30473)

wilsonge committed:
[4.0] Increase API Test Coverage (#26722)

* Add basic contact api tests

* Fix contact delete in the API

* Improve tests

* Update ContactModel.php

* Test for content categories

* Start work on tests for banners

* Move tests to bearer token approach

* Fix patch banner

* Fix category creation for non-content extensions

* Fix delete category

* Fix patch category

* State => published

* Don't hardcode catid for tests

* Debug

* Fix

* Fixes

* Fixes

* Append string

* Add catid to the update

* Fix auth

* Add workflow

* Trash article in the update

infograf768 committed:
[4.0] Messages/Alerts: using icons instead of text as heading (#30516)

hans2103 committed:
[4.0] com_finder ul instead of dl for easier styling (#30534)

hans2103 committed:
[4.0] [a11y] add statement on found results (#30535)

hans2103 committed:
add missing legend to fieldset (#30528)

brianteeman committed:
[4.0] Redundant words (#30555)

* [4.0] Redundant words

This is a repeat of #18852

Remove just from strings … it is a redundant word and is subjective.

> “Just” makes me feel like an idiot. “Just” presumes I come from a specific background, studied certain courses in university, am fluent in certain technologies, and have read all the right books, articles, and resources. “Just” is a dangerous word. http://bradfrost.com/blog/post/just/

Also removes easily and simply as they are also redundant and subjective

* comma

* Update plugins/sampledata/testing/language/en-GB/en-GB.plg_sampledata_testing.ini

Co-authored-by: Quy <quy@fluxbb.org>

* revert

Co-authored-by: Quy <quy@fluxbb.org>

david19161 committed:
[4.0] fix for inherit fields (#30557)

brianteeman committed:
[4.0] Accessibility plugin position (#30552)

This simple PR flips the display of the plugin from the right hand side to the left hand side in LTR languages and the opposite in RTL languages

This is to avoid clashing with the menu icon on mobile displays.

On desktops it probably makes more sense this way as its with the menus.

### Steps to reproduce the issue
Enable Plugin: System - Additional Accessibility Features
mobile view
scroll to bottom

### AFTER LTR

### AFTER RTL

brianteeman committed:
[4.0] Numbers not digits (#30559)

I have checked this with my client Cambridge University Press and they have checked the usage within the Corpus. https://en.wikipedia.org/wiki/Cambridge_English_Corpus The corpus is a multi billion word database of the usage of the english language in the world. The corpus is about how language is used and not just about the dictionary meaning

They wrote

Interesting! I’ve had a look and it looks like ‘number’ is more common – there are only a small number of examples with ‘digit’.

It’s hard to know for sure, as there are only very few examples of the exact sentences you’ve given so I did a more general search and often ‘number’ is used in sentences like ‘they could steal your credit card number and password’. But, I had a scan through and there are definitely lots of examples of e.g. ‘use a password that contains letters, numbers and special characters’, so I think you’re right to go with your feeling on this.

infograf768 committed:
[4.0] Useless installation lang strings (#30568)

drmenzelit committed:
[4.0] Display of horizontal mod_articles_news module (#30527)

* First step to make newsflash horizontal

* [4.0] Fix Alert double error display in Installation

* seperating multiple messages with border top

* Increase minimum length

* Fix meter displaying complete message prematurely

* Add strenthmeter attribute to installation

* Change meter values to display green when password requirements are met

* [4.0] {Cassiopea] Implementing Password length/meter

* [4.0] Fix Fieldset description display (#30435)

* [4.0] Url Language Code is not a prefix (#30440)

* [4.0] URL LANGUAGE CODE is not a prefix

* modified desc as suggested

* [4.0] Implementing display of fieldset descriptions for fieldset
children

* Taking off description

* scss cs

* use @Quy suggestion

* file cs

* [4.0] Media field fix (#30455)

* fix

* typo, (well nothing new)

* Allow media fields in subforms

- Cleanup Media field
- Expose initilazation method of Bootstrpa Modlas to a Joomla.Bootstrap object
- Initilise Modlas on the lifecycle of the Custom Element

* Paleolithic js mode only in the legacy…

* CS

* Proper check

* Pffff

* Update joomla-field-media.w-c.es6.js

* [4.0] Convert icon used for jgrid defaults to fas fa- (#30464)

* make icon-circle use fa-circle

* allows for including icon- in call

* move fa-fw to last class to fit convention (#30469)

* [4.0] Long labels wrapping (#30474)

When we have fields displayed in columns with the label displayed above the input there is a width limit of 240px on the label.

This results in long labels wrapping when it is not needed as the column is wider than 240px.

In english we dont see this is many places but the easiest is in the __configure edit screen_  fields

This simple PR changes the label width to 100% when the fields are displayed in columns like this.

As this is an scss change you will need to `node build.js --compile-css` in order to test

### Before

### After

* Fix mod_articles_latest (#30459)

* [4.0] Add a parameter for "back-to-top" button in Cassiopeia (#30441)

* [4.0] Add a parameter for back-to-top button in Cassiopeia

Added a parameter to show / hide the back-to-top button in Cassiopeia.
Currently the footer appears when a module has been published on position "footer". And the back-to-top only appears if the footer is visible. Since the footer is part of the grid, it is not easy to move the back-to-top outside the grid.

Now it is possible to decide to have a back-to-top or not. If yes, the footer will be rendered showing the button, independently if there is a module in the position footer.
It is also possible to show only modules (switch back-to-top off) or modules and back-to-top (switch on).

* Added smooth scrolling

Add smooth scrolling taking in account the preferences of the user (reduced motion).
Improve  language file.

* Correct height and position of back-to-top button

* Correct code style

* Correct code style 2

* Change smooth scrolling behavior to take in account reduced motion preferences

* Language file corrected, language string changed to match others

* Update language/en-GB/tpl_cassiopeia.ini

Link instead button

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update templates/cassiopeia/scss/blocks/_global.scss

Comment for smooth scroll

Co-authored-by: Brian Teeman <brian@teeman.net>

Co-authored-by: Brian Teeman <brian@teeman.net>

* fix improper family name. (#30481)

* [4.0] update helptoc (#30490)

* [4.0] update helptoc

Removes the string for the expired cache page that no longer exists and removes it from the index displayed on the left hand side in the help page

* api

* [4.0] Add new permissions-policy to the HTTPHeaders Plugin (#30491)

* add permissions-policy

* add Permissions-Policy to the dropdown

* alpha order

* [4.0] Error when changing status of tagged content items (#30466)

* Correct column name

* Fix changing UCM state

* [4.0] br tag (#30503)

* [4.0] br tag

In Joomla 4 we use `<br>` not `<br />`

* bad grep

* [4.0] Remove old search component (#30506)

* [4.0] Remove old search component

The regular search component is not in J4

This pr removes reference to it in the help system
`administrator/index.php?option=com_admin&view=help`

* revert

* [4.0] Help Dashboard (#30508)

* [4.0] Help Dashboard

This PR makes multiple changes to the help dashboard. In order of importance they are.

- Created a new section "Start Here"
- Moved "Joomla Help" to the "Start Here" section to give it greater priority as it really is the first place you should look for help
- Removed target=_blank from "Joomla Help"
- Moved "Documentation Wiki" from "Resources" to the top of "Additional Help"

* deduplicate

* compass icon

* Moved css for mod_articles_news to media folder

User webAsset Manager to load specific css for module

* Move web asset call to horizontal.php

Co-authored-by: Jean-Marie Simonet <infografjms@gmail.com>
Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: dGrammatiko <d.grammatiko@gmail.com>
Co-authored-by: Bear <programming@hallhome.us>
Co-authored-by: Brian Teeman <brian@teeman.net>
Co-authored-by: Christiane Maier-Stadtherr <dev@chmst.de>
Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>
Co-authored-by: SharkyKZ <sharkykz@gmail.com>

infograf768 committed:
[4.0] [Multilingual Status module] Adding displaying a possible error if URL Language Code is empty (#30537)

* [4.0] [Multilibgual Status module] Adding displaying a possible error if
sef is not set

* Simplify 2 strings

astridx committed:
[4.0] Update Readme for Api tests (#30539)

dgrammatiko committed:
[4.0] favicon changes to support child templates (#30388)

dgrammatiko committed:
[4.0] Child templates consistency (#30387)

maliouris committed:
Fix media manager on 0 maximum file size (#30572)

Co-authored-by: Ioannis Maliouris <giannism@e-avenue.eu>

maliouris committed:
Show exception messages on media error (#30573)

* Show exception messages on media error

* Add COM_MEDIA_ERROR_WARNFILETOOLARGE to languages and default texts

* Show exception message without tranforming in uppercase

* Remove spaces from text.es6

Co-authored-by: Ioannis Maliouris <giannism@e-avenue.eu>

zero-24 committed:
apply the proposed changes from @brianteeman to the testing plugin language file (#30571)

N6REJ committed:
[4.0] Align content info block icons (#30543)

* convert if (icon) to switch (icon)

* revert to using mapped fix.

* multi-reverts

* make icons even width

* reverted improper change.

pdkpavankumar committed:
[4.0] fix color selector box too small issue (#30585)

* [4.0] fix color selector box too small issue

* removed redundant & for fixing sass lint error

Commit:
Fix preset key. (#30583)

pdkpavankumar committed:
dropdown corner css issue fix (#30596)

* dropdown corner css issue fix

* fixing phpcs error

* removing whitespace

* Apply suggestions from code review

Co-authored-by: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

infograf768 committed:
[4.0] Modify display and tip for Edit Article in Frontend (#30588)

* [4.0] Modify display and tip for EDit Article in Frontend

* take off commented line

* correcting wrong a11y parts

* various improvements

* cs

* cs

Fedik committed:
[4.0] Fix "explain" feature in the debug plugin (#30580)

* Fix "explain" feature in the debug plugin

* Explain for PostgreSQL

* Update build/media_source/plg_system_debug/widgets/sqlqueries/widget.es5.js

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update rendering of Explain table

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

shim-sao committed:
[4.0] Saving article cause : hits = '' is not a valid integer (#30523)

* [4.0] fix Templates : Edit style - unsaved menu assignments

* Correct variable $userId not $userid in save function

* Update administrator/components/com_templates/src/Model/StyleModel.php

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

* Update administrator/components/com_templates/src/Model/StyleModel.php

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

* Update administrator/components/com_templates/src/Model/StyleModel.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_templates/src/Model/StyleModel.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* [4.0] Resizing template editor causes multiple editor instances #22778

Tested also by clicking on add readmore and other actions

* Saving article cause : '' is not a valid integer

Default hits value should be a valid inetger to be saved in the database.

* Add last blank line

Add the last blank line

Co-authored-by: SharkyKZ <sharkykz@gmail.com>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

infograf768 committed:
[4.0] Adding back string deleted by error in 30588 (#30601)


* also take off useless role button when item is checked out

hans2103 committed:
[4.0] [JLayout] merge two equal JLayouts for iconclass into one new JLayout for iconclass (#30470)

* :art: merge two equal JLayouts into one new JLayout

* replace hard coded icon to call for JLayout

* replace icon mapping to call for JLayout

* add icon protected to layout

* icon protected moved to jlayout iconclass

* remove fas

* add icon trash to mapping

* add parameter to check if html should be rendered or icon class only

* simplify php

* replace massive if-then-elseif-then by switch

* merge duplicate outcome fa-times

* be consistent

* simplify switch thx @quy

* copy changes from https://github.com/joomla/joomla-cms/pull/30468/files

* copy changes from https://github.com/joomla/joomla-cms/pull/30468/files

* remove setting icon to itself

* copy changes from https://github.com/joomla/joomla-cms/pull/30468

* change to use layout

* Update libraries/src/HTML/Helpers/ActionsDropdown.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* use LayoutHelper in function

* let default html output be true, instead of false

* Update layouts/joomla/icon/iconclass.php

I like this PHP7 operator

Co-authored-by: Quy <quy@fluxbb.org>

Co-authored-by: Troy T. Hall <programming@hallhome.us>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Quy <quy@fluxbb.org>

pdkpavankumar committed:
[4.0] adding additional check to avoid js script error (#30593)

* adding additional check to avoid js script error

* fixing js lint error

richard67 committed:
[4.0] Revert removal of the "custom_data" field of the "extensions" database table. (#30373)

wojsmol committed:
Fix help link (#30594)

infograf768 committed:
[4.0] Modify display and tip for Edit Contact in Frontend (#30605)

* [4.0] Modify display and tip for Edit Contact in Frontend

* Update administrator/components/com_contact/src/Service/HTML/Icon.php

Co-authored-by: Quy <quy@fluxbb.org>

* Update Icon.php

Co-authored-by: Quy <quy@fluxbb.org>

ruchikanagarkar committed:
[4.0] save group button hover fix (#30618)

Commit:
Get categories from component (#30613)

zero-24 committed:
set a default site name when the option is empty (#30520)

richard67 committed:
Fix order of options in Cassiopeia template (#30629)

N6REJ committed:
Align icons (#30632)

Aligns icons in mod_login

Commit:
Fixing com_media URLs for frontend (#30617)

obuisard committed:
Update template.js (#30641)

There is a typo in the Ajax request 'url' parameter.

Commit:
Remove unused model (#30600)

wilsonge committed:
[4.0] Fix extension SQL check for generic components, modules and plugins (#30626)

Commit:
Store memory and time as float (#30574)

hans2103 committed:
Feature/4.0 dev a11y  com finder refactor search result (#30550)

alikon committed:
fix unpublished contact (#30642)

alikon committed:
[4.0] remove user from user group (#30355)

wilsonge committed:
Bump to 4.0.0 beta 4

infograf768 committed:
[4.0] List Contacts in a Category display

wilsonge committed:
Bump back to devel

infograf768 committed:
a11y

alikon committed:
[4.0] re-throw the original exception (#30653)

* re-throw the original exception

* re-throw the original exception

infograf768 committed:
added font-weight to dd

infograf768 committed:
small css change

Commit:
Missing string when uploading files (#30671)

trananhmanh89 committed:
fix get list error (#30672)

ceford committed:
In Joomla 4 - Rename Control Panel as Home Dashboard (#30673)

wilsonge committed:
Bump database package and system tests

wilsonge committed:
Update system tests again

Ruud68 committed:
Fix tabs not wrapping and trigger horizontal scrolling (#30675)

hans2103 committed:
Add missing space between attributes (#30681)

infograf768 committed:
[4.0] Missing alerts strings (#30684)

Creating the missing strings in front and backend
JCLOSE="Close"
JOK="OK"
JOPEN="Open"

Added the declarations in message.php

Commit:
[4.0] Fix toolbar collapse in mobile view (#30131)

* Fix toolbar collapse in mobile view

* Match toolbar collapse with cog icon

* Removed unused constant

SharkyKZ committed:
Correct field name (#30687)

SharkyKZ committed:
Fix media URLs when SEF suffix is enabled (#30688)

brianteeman committed:
[4.0] Batch fieldset markup [a11y] (#30694)

A fieldset should have a legend not a label. There is no visible change its just in the markup which is now valid.

richard67 committed:
Dont use id in WHERE clause for updating menu items in update SQL script

richard67 committed:
Fix Atum sidebar wrapper collapse on resizing mobile view (#30686)

brianteeman committed:
[4.0] Admin menu redundant link (#30696)

* [4.0] Admin menu redundant link

When adjacent links go to the same location this results in additional navigation and repetition for keyboard and screen reader users. WCAG 2.4.4

The parent menu item for Smart Search is the same as the first child  menu item.

Although the first one is disabled by the js and only acts as means to expand the submenu its still a problem for some assistive technologies.

This PR follows the pattern of all other core components and changes the parent link to index.php?option=com_finder

This can be tested either by performing a clean installation or by manually applying the sql in the update script.

Note that "database fix" will not apply the change in the update script as it is a content change not a structural change.

* Update administrator/components/com_admin/sql/updates/mysql/4.0.0-2020-09-19.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update administrator/components/com_admin/sql/updates/postgresql/4.0.0-2020-09-19.sql

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.0] more elements hard coded to string (#30720)

infograf768 committed:
[4.0] Cassiopea: fix category x layout (#30710)

brianteeman committed:
[4.0] Admin Module Help strings (#30458)

brianteeman committed:
[4.0] codestyle (#30724)

* [4.0] codestyle

fix typo in code comments - code review only required

* spelling

richard67 committed:
Add update SQL script for previous 4.0 beta versions

SharkyKZ committed:
Move Route::_() calls to layout (#30733)

brianteeman committed:
[4.0] Composer update 19 sep (#30692)

Package operations: 0 installs, 38 updates, 0 removals
  - Updating symfony/deprecation-contracts (v2.1.3 => v2.2.0): Downloading (100%)
  - Updating joomla/database (2.0.0-beta2 => 2.0.0-beta3): Downloading (100%)
  - Updating laminas/laminas-zendframework-bridge (1.0.4 => 1.1.1): Downloading (100%)
  - Updating laminas/laminas-diactoros (2.3.1 => 2.4.1): Downloading (100%)
  - Updating symfony/string (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/service-contracts (v2.1.3 => v2.2.0): Downloading (100%)
  - Updating symfony/console (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/var-dumper (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/error-handler (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/options-resolver (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/ldap (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/web-link (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/yaml (v5.1.3 => v5.1.5): Loading from cache
  - Updating squizlabs/php_codesniffer (3.5.5 => 3.5.6): Loading from cache
  - Updating symfony/finder (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/event-dispatcher-contracts (v2.1.3 => v2.2.0): Downloading (100%)
  - Updating symfony/event-dispatcher (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/css-selector (v5.1.3 => v5.1.5): Loading from cache
  - Updating phpdocumentor/type-resolver (1.3.0 => 1.4.0): Downloading (100%)
  - Updating phpdocumentor/reflection-docblock (5.2.0 => 5.2.2): Downloading (100%)
  - Updating codeception/lib-asserts (1.12.0 => 1.13.1): Loading from cache
  - Updating codeception/codeception (4.1.6 => 4.1.7): Loading from cache
  - Updating codeception/module-asserts (1.2.1 => 1.3.0): Loading from cache
  - Updating codeception/module-rest (1.2.1 => 1.2.3): Downloading (100%)
  - Updating symfony/process (v5.1.3 => v5.1.5): Loading from cache
  - Updating codeception/module-webdriver (1.1.0 => 1.1.1): Loading from cache
  - Updating composer/semver (1.5.1 => 1.7.0): Downloading (100%)
  - Updating composer/xdebug-handler (1.4.2 => 1.4.3): Loading from cache
  - Updating doctrine/annotations (1.10.3 => 1.10.4): Loading from cache
  - Updating symfony/filesystem (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/stopwatch (v5.1.3 => v5.1.5): Loading from cache
  - Updating joomla/filesystem (1.5.2 => 1.6.0): Loading from cache
  - Updating composer/ca-bundle (1.2.7 => 1.2.8): Loading from cache
  - Updating nyholm/psr7 (1.3.0 => 1.3.1): Downloading (100%)
  - Updating spomky-labs/base64url (v2.0.2 => v2.0.3): Loading from cache
  - Updating symfony/dom-crawler (v5.1.3 => v5.1.5): Loading from cache
  - Updating symfony/browser-kit (v5.1.3 => v5.1.5): Loading from cache
  - Updating brick/math (0.8.15 => 0.8.17): Loading from cache

### note
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.

Fedik committed:
[4.0] Fix Debug plugin to display query parameters (#30717)

Ruud68 committed:
Fix hiddenLabel and add hiddenLegend form attribute (#29710)

Commit:
[4.0] Cleanup cassiopeia chromes and rename "default" to "card" (#30729)

* Renaming the "default" chrome to "card" as it isn't a default chrome at all.

* There is no "xhtml" chrome anymore

SharkyKZ committed:
[4.0] Fix saving images (#30730)

* Fix saving images

* Typo

* CS

SharkyKZ committed:
[4.0] Remove use of ReflectionParameter::getClass() (#30581)

Harmageddon committed:
Fixed data type.

brianteeman committed:
[4.0] spelling (#30742)

fix my own mistake the correct spelling is fulfil

SharkyKZ committed:
Use renderField() method to render fields (#30738)

Commit:
Removing card.scss overrides as we don't use those classes

SharkyKZ committed:
[4.0] Check out improvements related to nullable columns (#30747)

* Show 0 values as checked out in nullable columns

* Check in 0 values in nullable columns

* Change comparison

hans2103 committed:
Remove obsolete html code (#30737)

Commit:
Remove the chrome "cardGrey". The same effect can be achieved by using the module class "card-grey" with the "card" chrome (#30734)

Commit:
Fix icons not displaying (#30749)

Commit:
Improve batch text (#28447)

brianteeman committed:
[4.0] Cassiopeia missing string (#30765)

The string is used in the template style page of the template

hans2103 committed:
[4.0] Wrap all buttons in btn-group to improve styling (#30761)

* wrap all buttons in btn-group to improve styling

* add if-then statement to hide button if not needed

hans2103 committed:
[4.0] Add missing Table Caption (#30763)

* Add missing Table Caption

* Update components/com_content/tmpl/category/default_articles.php

Co-authored-by: Brian Teeman <brian@teeman.net>

* add missing language string

Co-authored-by: Brian Teeman <brian@teeman.net>

pdkpavankumar committed:
[4.0] Fancy selectbox fix (#30739)

* fixing fancy select box alignment

* Trigger notification

* implemented code review comments

* updating margin from 45px to 50px

* Trigger notification

infograf768 committed:
[4.0] Modifying com_actionlogs string (#30758)

alikon committed:
[4.0][CLI] com_finder use console command (#30768)

brianteeman committed:
[4.0] Template layout select (#30772)

If the label is "Fluid Layout" then the options should be "Yes/No"

But thats not very descriptive so this PR instead changes the label to "Layout" and keeps the options to "Fluid/Static"

chmst committed:
Improve code, remove separator (#30785)

brianteeman committed:
[4.0] mod_article_news readmore (#30780)

This PR simply changes the Article News module to use the existing joomla.content.readmore layout

richard67 committed:
Update package-lock.json (#30713)

zero-24 committed:
[4.0] Pluginless lazyloading for the core (#30748)

* remove the plugin code

* make sure there is no error message on attributs passed as string

* implement width & height into the media field

* add width & hight to com_media

* add width & height to the file list

* fix file path

* add width & heigth to the media field

* there is no alt text comming from com_media yet

* patch JS thanks @dgrammatiko

* Update installation/sql/postgresql/base.sql

Co-authored-by: Quy <quy@fluxbb.org>

* Update installation/sql/mysql/base.sql

Co-authored-by: Quy <quy@fluxbb.org>

* fix issues mention vy @dgrammatiko and implement width and heigth in the intro and full image layout

* When we have no image nothing is going to be displayed

* move json_decode & change back to the original image url

* take off px for width and height

* reset other changes that has been moved to #30784

* also reset the image list layout

* Update libraries/src/HTML/HTMLHelper.php

* Add update SQL scripts 4.0.0-2020-09-27.sql (#47)

Co-authored-by: Quy <quy@fluxbb.org>
Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

drmenzelit committed:
Change grid minmax definition for newsflash module (#30781)

Change grid minmax definition for newsflash module to better fit in small displays
See issue https://github.com/joomla/cassiopeia/issues/119

chmst committed:
[4.0] Blog view links (#30788)

* links in blogs views as oredered list

* equal layouts in category and featured

pdkpavankumar committed:
[4.0] Cassiopea table css (#30740)

* fixing hardcoded value to variable value for cassiopea table css

* adding retun carriage

* removed the override paddings

SharkyKZ committed:
[4.0] Error in legacy plugins when method contains $event argument (#30575)

* Error in legacy plugins when method contains $event argument

* Unnecessary parentheses

* Correct comment

* Optimize

SharkyKZ committed:
Replace file_exists() where applicable (#30800)

zero-24 committed:
Update libraries/src/Application/CMSApplication.php

SharkyKZ committed:
Clean up com_users forms (#30796)

brianteeman committed:
[4.0] Workflow missing string (#30811)

PR for #30807

### Steps to reproduce the issue

Enable com_workflow in Backend, open an Article in Frontend, open Tab 'Publishing'

### Expected result
Translated Constant.

### Actual result
Constant

brianteeman committed:
change layout to use its own param "link_intro_image"

brianteeman committed:
new config option

brianteeman committed:
global string

brianteeman committed:
move param to blog/featured layout tab

brianteeman committed:
menu option blog category menu option featured

brianteeman committed:
update layout to use the new param AND fixing the missing noauth check and fix the missing title on the link

brianteeman committed:
sync api languge file

brianteeman committed:
installation sql

drmenzelit committed:
[4.0] Add css style for mod_articles_news vertical layout (#30797)

tecpromotion committed:
Update com_mails.ini (#30842)

delete a duplicate string in com_mails

brianteeman committed:
[4.0] hidden-xs-up (#30837)

brianteeman committed:
[4.0] tinmyce toolbar icons (#30825)

The icons used in the tinymce configuration plugin come from a different place than the icons used in the editor itself (I don't know why).

The icons used in the configuration plugin for visualchars and visualblocks were out of date.

brianteeman committed:
[4.0] Language reinstall (#30810)

Adds a visual indicator to the button for reinstalling a language by changing the button color

PR for #30808

brianteeman committed:
[4.0] Article Links markup (#30824)

Changes the class on the list of article links to be consistent with other classes and to enable them to be styled uniquely by the template

infograf768 committed:
[4.0] Site Contacts Category List Search Filters (#30813)

infograf768 committed:
[4.0] Missing js file in #30813 (#30859)

* [4.0] Missing js file in #30813

* missing new line

tecpromotion committed:
[4.0] Update lib_joomla.ini (#30867)

* Update lib_joomla.ini

* fix

thanks @infograf768

brianteeman committed:
[4.0] com_media options (#30852)

Remove duplicated warning messages

Solves #29309

brianteeman committed:
[4.0] joomla update show password (#30849)

* [4.0] joomla update show password

PR for #29765

### Steps to reproduce the issue
Joomla 4
System -> Joomla (Update box)
Choose any file (#29763 bug allows any file)
Click upload and install

### Expected result
Password hide/reveal icon is clickable like all other password boxes in Joomla 4

* icon width

SharkyKZ committed:
[4.0] Fall back to system template when template file not found (#30857)

* Fall back to system template

* Load parent language file only when needed

* Correct template file

SharkyKZ committed:
Namespace (#30865)

brianteeman committed:
[4.0] Templates: Styles (Administrator) menu color changes on save (#30847)

PR for #29958

The values in the XML are different to those in the variables.scss when they should be the same

As no reply from @coolcat-creations or @angieradtke to confirm which is the correct version I took the choice to make the xml match the scss

infograf768 committed:
[4.0] Fix List Contacts Category List Limit

brianteeman committed:
[4.0] xtd-editor button for module (#30892)

* [4.0] xtd-editor button for module

The icon used was the same as the one for articles. This PR changes it to be the cube that is used in the module manager.

* cube

Commit:
[4.0] Fix pagebreak modal missing style (#30889)

brianteeman committed:
[4.0] treeselect alerts (#30854)

PR for #29489

brianteeman committed:
[4.0] sticky sidebar menu (#30853)

On very long pages (eg install from web) as you scroll down the page the sidebar menu disappears off screen.

This PR makes the sidebar sticky so that it never scrolls out of the screen just as the subhead toolbar is sticky.

pr for #30019

brianteeman committed:
[4.0] Media Manager close preview (#30851)

* [4.0] Media Manager close preview

* suggestion from @fedik

brianteeman committed:
[4.0] Change the size of the template name font in module position selector (#30890)

PhilETaylor committed:
Add generic AJAX error message translation (#30909)

infograf768 committed:
Using getUserStateFromRequest

brianteeman committed:
[4.0] media manager transparent preview (#30870)

The media manager uses a background of checked squares in the thumbnail view which is useful when you have a transparent image such as the joomla_black.png

However in the preview it uses a solid white background which hides most of this image.

This pr replaces the white background with the checked squares

Commit:
Joomla\CMS\Installation\Helper\ArrayHelper not found (#30901)

[4.0b4] Class 'Joomla\CMS\Installation\Helper\ArrayHelper' not found
see issue #30900

zero-24 committed:
make sure hsts is only added on https sites (#30902)

brianteeman committed:
[4.0] resize webauthn logo (#30846)


PR for #30180

No need to load a 200kb image when its only being displayed on a button

brianteeman committed:
[4.0] Button text double translation (#30935)

Enable language debug
Open an article
On the save button dropdown you will see `??**Save to Menu**??`

Apply the PR
Now the button is correctly displayed as `**Save to Menu**`

ceford committed:
Change all JHELP value strings in jooml.ini (#30712)

* Changed all JHelp file name strings

* Fixed Admin_Modules:_Frontend_Link in joomla.ini

* Changed JHelp module admin strings to new file names

* Made api joomla.ini JHELP same as joomla.ini

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

shur committed:
Update ArticlesNewsHelper.php (#30946)

misprint 'Filer' > 'Filter'

infograf768 committed:
[4.0] Normalise Featured Contact List (#30868)

* taking off NUM column

* adding back itemscope

* string used in form

* Getting rid of fieldset

* removing title as required

infograf768 committed:
[4.0] Frontend List articles in a category: adding Clear filter,
refactoring layout for a11y+

infograf768 committed:
[4.0] Frontend: Normalise List contacts

brianteeman committed:
[4.0] yubikey text (#30879)

don't display the heading additionally in the body

Commit:
Remove None Selected option (#30987)

richard67 committed:
Add update SQL to delete plg_authentication_gmail from extensions table (#30941)

SharkyKZ committed:
[4.0] Small refactor of typehint check in CMSPlugin (#30907)

* Refactor typehint check

* Make private

infograf768 committed:
[4.0] Ooops: frontend contacts layouts filter-field correction (#30999)

dgrammatiko committed:
No ref to non existing elements (v-if) (#30917)

PhilETaylor committed:
Remove Unnecessary Local Variable

brianteeman committed:
[4.0] codemirror height (#30990)

SharkyKZ committed:
Correct @deprecated version (#31018)

SharkyKZ committed:
Update comment (#31019)

brianteeman committed:
[4.0] template editor tree (#31029)

Adds a small margin top and bottom so that the top and bottom shadow is visible

Commit:
Increase right margin (#31022)

SharkyKZ committed:
Missing field class (#31015)

Fedik committed:
No hover (#30993)

SharkyKZ committed:
Add WebP support

SharkyKZ committed:
Add tests

SharkyKZ committed:
Update image

PhilETaylor committed:
Remove unnecessary semicolon (#31055)

PhilETaylor committed:
Cleanup of switch statements to remove code smell

PhilETaylor committed:
Update TinyMCE Links to documentation (#31053)

Denitz committed:
Prioritise /tmpl component folder for menu items .xml definitions and componentlayout field (#30983)

PhilETaylor committed:
typo (#31062)

PhilETaylor committed:
[4] Return value type is not compatible with declared type in doc (#31059)

* Return value type is not compatible with declared type in doc

* add null return type

* update return type

PhilETaylor committed:
s/Santize/Sanitize (#31069)

PhilETaylor committed:
remove three unneeded semicolons (#31070)

PhilETaylor committed:
Return is unnecessary as the last statement in a void method (#31072)

SharkyKZ committed:
Reduce indentation

Co-authored-by: Quy <quy@fluxbb.org>

SharkyKZ committed:
Update comment

Co-authored-by: Quy <quy@fluxbb.org>

infograf768 committed:
[4.0] Take of the Trash dropdown Action when filtering by trashed items

brianteeman committed:
[4.0] ATUM prefers-reduced-motion [a11y] (#30873)

Applies a global css rule to support users who have set prefers-reduced-motion in their operating system.

Toggle the menu sidebar
If working correctly it will be a smooth transition if animations are on and it will be an immediate close if animations are off

Origin of the chosen css
https://web.dev/prefers-reduced-motion/#(bonus)-forcing-reduced-motion-on-all-websites

ceford committed:
Added Mail template Help strings (#31054)

HLeithner committed:
Ignore missing template description (#31047)

PhilETaylor committed:
[4] fix obviously wrong docs (#31064)

* docs

* more obviously wrong docs

PhilETaylor committed:
[4] Ternary Expression can be safely replaced with condition removing codesmell (#31071)


Co-authored-by: Harald Leithner <leithner@itronic.at>

infograf768 committed:
[4.0] Frontend: correcting Tags layouts (#30998)


Co-authored-by: Quy <quy@fluxbb.org>

PhilETaylor committed:
Update administrator/components/com_installer/src/Model/InstallerModel.php

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

PhilETaylor committed:
Update administrator/components/com_installer/src/Model/InstallerModel.php

Co-authored-by: SharkyKZ <sharkykz@gmail.com>

Commit:
Hide toolbar since no options available

infograf768 committed:
[4.0] Fix Tags Compact list bug (#31093)

SharkyKZ committed:
[4.0] Expired content shown in tag view (#31097)

* Expired content shown in tag view

* Use strict

chmst committed:
[4.0] Replace old contact icons by icons (#30985)

* Add Icon Webpage to contact config

* add webpage to the list, add alt-texts to images, remove the old images

* if no image is set, use an icon instead

* add an icon for msc information

* remove old icons for contact

* remove spaces, indent

* phpcs shoter lines, phpcs

* replace empty(..) by !(   )

* Update components/com_contact/tmpl/contact/default_address.php

Co-authored-by: Quy <quy@fluxbb.org>

* Update components/com_contact/tmpl/contact/default_address.php

Co-authored-by: Quy <quy@fluxbb.org>

* Update components/com_contact/tmpl/contact/default_address.php

Co-authored-by: Quy <quy@fluxbb.org>

Co-authored-by: Quy <quy@fluxbb.org>

brianteeman committed:
[4.0] Extra space cpanel tables (#31030)

Removes the margin bottom from tables included in cpanel cards.

PhilETaylor committed:
Add missing import for exception (#31081)

Add missing import so that "Exception\ResourceNotFound" is correctly resolved.

SharkyKZ committed:
[4.0] Move calls out of the loop (#31080)

* Move calls out of the loop

* Combine pathinfo() calls

SharkyKZ commented:
Can you do it the same as in `error.php`?

https://github.com/joomla/joomla-cms/blob/f4b26449d5020a3ddb1f909c0f21b232212b561c/templates/cassiopeia/error.php#L28-L29

Otherwise `$pageclass` is undefined when no menu item.

richard67 commented:
@SharkyKZ This PR here changes error.php, too, and I don't understand why. It was ok like it was before, or not? Am I missing something?

richard67 commented:
@sakiss Please revert your change in error.php. It was good like it was before, using the ternary expression, and you now introduced an error with undefined $pageclass when there is no menu item. And as @SharkyKZ suggested please change index.php so it uses the same ternary expression as error.php does without your change.

PhilETaylor committed:
Variable key is used in both the inner and outer foreach loops (#31066)

Variable key is used in both the inner and outer foreach loops which could cause issues (it doesn't but it could, and is code smell reported by phpStorm)

PhilETaylor committed:
Return value type is not compatible with declared (#31061)

Return value type is not compatible with declared type in PHPDoc block

Hackwar committed:
[4.0] Non-empty, non-matching searches shouldn't return anything (31040)

This fixes #30967. When empty searches are allowed and the search is non-empty, but not matching, then it shouldn't return any results.

PhilETaylor committed:
[4] Update PHPDoc comment to match method signatures in installation app (#31060)

* Update PHPDoc comment to match method signatures in installation app

* Update PHPDoc comment to match method signatures in installation app

* code style

* Update installation/src/Application/InstallationApplication.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update installation/src/Application/InstallationApplication.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

* Update installation/src/Application/InstallationApplication.php

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>

brianteeman committed:
[4.0] Share image link (#30855)

* [4.0] Share image link

Steps to reproduce the issue
Go to media manager in admin, select an image and select to get the shareable link.
click "Get shareable link"

Expected result
When clicking the copy button, the left border doesn't disappear

Actual result
left border of the copy button disappears on click, and on any click/selection of the actual url

* cs

PhilETaylor committed:
Remove unused array - Array is only updated, but never queried (#31057)

Fedik committed:
Correcting addInline() to support WebAssetItemInterface and String, as planed (#30986)

Harmageddon committed:
Fixed frontend module editing for non-superusers. (#30779)

Commit:
Fix paths (#31102)

Fix paths in codeowners file.

richard67 commented:
@sakiss And in addition to the above comment, please update the branch of this PR to the latest 4.0-dev branch of the CMS repository, not of your fork. It seems your 4.0-dev branch was very outdated when you have created this PR and the other one, #31099 based on it, because both PRs fail in our automated test due to outdated testing configuration from before February.

Let me know if you need help with.

Thanks in advance.

PhilETaylor committed:
Remove unused namespace declaration (#31007)

Hackwar committed:
Fixing "expand all" for com_finder (#31073)

PhilETaylor committed:
Return is unnecessary as the last statement in a void method (#31051)

wilsonge commented on templates/cassiopeia/error.php:
```diff
@@ -20,14 +20,18 @@
-$itemid   = $app->input->getCmd('Itemid', '');
-$sitename = htmlspecialchars($app->get('sitename'), ENT_QUOTES, 'UTF-8');
-$menu     = $app->getMenu()->getActive();
-$pageclass = $menu !== null ? $menu->getParams()->get('pageclass_sfx', '') : '';
```
How is this different to the ternary that was here before?

richard67 replied:
As @SharkyKZ and me already pointed out, the difference is that with this PR, the `$pageclass` variable will be undefined in case if the `$menu` variable is null, while with the ternary it will be an empty string. So it is already clear that this change in error.php here has to be reverted completely, and that in index.php the ternary should be used, too, which is not the case in current 4.0-dev and so causes the issue which this PR here tries to fix in the wrong way.

It's just on the author to react to that.

SharkyKZ replied:
`$pageclass` is now defined outside of `$menu` check so the code is working fine at least.

sakiss replied:
> the $pageclass variable will be undefined in case if the $menu variable is null,

No is not.
The only reason for changing that in error.php is to be in accordance with index.php.
I can change that to use ternary but the logic is quite the same. I don't think is the wrong way but rather a matter of taste. 
As for the repo update, i am having hard times, running the `npm update` and opened an issue https://github.com/joomla/joomla-cms/issues/31114, any help is welcome.

richard67 replied:
Yes, please revert changes in error php because they are unnecessary, the code was good also before this PR, and change index.php so it uses the same ternary as error.php does.

infograf768 committed:
no need to check again code.edit.state

SharkyKZ committed:
[4.0] Fix retrieving articles for contact (#31107)

richard67 commented:
Added the "Updates Requested" label so it's clear for testers that this PR needs changes as specified in review comments above.

rdeutz committed:
Update HtmlView.php

cs fix

sakiss committed:
Use of ternary instead of if

sakiss commented:
I am closing this due to conflicts. Will open a new one.
